File: ccif/src/java/other/org/himalay/ccif/FirmWareDate.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class FirmWareDate extends BinStruct implements PublicBinMsg{




    // members variables
    // year
    public int year ;
    // month
    public short month ;
    // date
    public short date ;

        public FirmWareDate () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize year
        
        // Initialize month
        
        // Initialize date
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read year
        {year=istream.readUnsignedShort(); retVal+=2;}
        // read month
        {month=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read date
        {date=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
        
    
                // write year
        ostream.writeShort(year); retVal +=2;
        // write month
        ostream.writeByte(month); retVal +=1;
        // write date
        ostream.writeByte(date); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("FirmWareDate\n");
    dc.increaseIndent();
        int retVal= 0;
        // write year
        dc.indent();dc.getPs().println("year="+year+"(0x"+ Integer.toHexString(year) + ")" );
        // write month
        dc.indent();dc.getPs().println("month="+month+"(0x"+ Integer.toHexString(month) + ")" );
        // write date
        dc.indent();dc.getPs().println("date="+date+"(0x"+ Integer.toHexString(date) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for year
    //public int getYear()
    //{
    //    return year ;
    //}

    
    // Setter for year
    //public void setYear(int val)
    //{
    //    this.year= val;
    //}
    // Getter for month
    //public short getMonth()
    //{
    //    return month ;
    //}

    
    // Setter for month
    //public void setMonth(short val)
    //{
    //    this.month= val;
    //}
    // Getter for date
    //public short getDate()
    //{
    //    return date ;
    //}

    
    // Setter for date
    //public void setDate(short val)
    //{
    //    this.date= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/FirmWareDate.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "FirmWareDate.h"

#ifdef org_himalay_ccif_FirmWareDate__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
FirmWareDate::FirmWareDate() {
    references= 0;


    // year
    year= (ui16)0;
    // month
    month= (ui8)0;
    // date
    date= (ui8)0;
}
// Destructor
FirmWareDate::~FirmWareDate() {
    // year

    // month

    // date

}

int FirmWareDate::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // year
    {
        year=istream.readUI16();
        retVal+=2;
    }
    // month
    {
        month=(istream.readUI8());
        retVal+=1;
    }
    // date
    {
        date=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int FirmWareDate::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read year
    {
        year=istream.readUI16();
        retVal+=2;
    }
    // read month
    {
        month=(istream.readUI8());
        retVal+=1;
    }
    // read date
    {
        date=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int FirmWareDate::write(DataOutputStream& ostream) { // throws IOException




    int retVal= 0;
    // write year
    ostream.writeUI16(year);
    retVal +=2;
    // write month
    ostream.writeUI8(month);
    retVal +=1;
    // write date
    ostream.writeUI8(date);
    retVal +=1;

    return retVal;
}

int FirmWareDate::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("FirmWareDate\n");
    dc.increaseIndent();
    int retVal= 0;
    // write year
    dc.indent();
    dc.getPs().print("year=");
    dc.getPs().println((long)year);
    // write month
    dc.indent();
    dc.getPs().print("month=");
    dc.getPs().println((long)month);
    // write date
    dc.indent();
    dc.getPs().print("date=");
    dc.getPs().println((long)date);
    dc.decreaseIndent();
    return retVal;
}


// Getter for year
//ui16 FirmWareDate::getYear()
//{
//return this->year;
//}

// Setter for year
//void FirmWareDate::setYear(ui16 val)
//{
//this->year= val;
//}

// Getter for month
//ui8 FirmWareDate::getMonth()
//{
//return this->month;
//}

// Setter for month
//void FirmWareDate::setMonth(ui8 val)
//{
//this->month= val;
//}

// Getter for date
//ui8 FirmWareDate::getDate()
//{
//return this->date;
//}

// Setter for date
//void FirmWareDate::setDate(ui8 val)
//{
//this->date= val;
//}

int FirmWareDate::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_FirmWareDate__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::FirmWareDate* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::FirmWareDate* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code